### PR TITLE
build: migration and max withdraw updates

### PR DESCRIPTION
**Pull request by Schlagonia**

### CODE REVIEW

File: contracts/BaseStrategy.sol [hunk 1]
```diff
@@ -65,37 +76,37 @@ abstract contract BaseStrategy {
...
+
     function withdraw(
         uint256 amount,
         address receiver,
         address owner
-    ) public returns (uint256) {
-        require(msg.sender == vault && msg.sender == receiver, "not owner");
-        require(amount <= maxWithdraw(vault), "withdraw more than max");
+    ) public onlyVault returns (uint256) {
+        require(amount <= _maxWithdraw(vault), "withdraw more than max");
```
Comment (jmonteer) [inline contracts/BaseStrategy.sol]: use msg.sender instead of vault

File: contracts/BaseStrategy.sol [hunk 2]
```diff
@@ -65,37 +76,37 @@ abstract contract BaseStrategy {
...
 
-        uint256 amountWithdrawn = _withdraw(amount, receiver, owner);
-        IERC20(asset).transfer(receiver, amountWithdrawn);
+        uint256 amountWithdrawn = _withdraw(amount);
+        IERC20(asset).transfer(vault, amountWithdrawn);
```
Comment (jmonteer) [inline contracts/BaseStrategy.sol]: use msg.sender instead of vault

**Review by jmonteer — CHANGES_REQUESTED** [review]

File: contracts/BaseStrategy.sol (hunk 1 above)
Comment (Schlagonia) [reply contracts/BaseStrategy.sol]: https://github.com/jmonteer/yearnv3-strat-compoundv3-lender/pull/6/commits/c4637d7efa34f5f6c1b5b170cb39ac98001ec514

File: contracts/BaseStrategy.sol (hunk 2 above)
Comment (Schlagonia) [reply contracts/BaseStrategy.sol]: https://github.com/jmonteer/yearnv3-strat-compoundv3-lender/pull/6/commits/c4637d7efa34f5f6c1b5b170cb39ac98001ec514

**Review by jmonteer — APPROVED** [review]

**Pull request merged**